tests/backup/backup_test.go

```diff
@@ -334,6 +332,511 @@ var _ = Describe("{BackupCreateKillStoreRestore}", func() {
+				CreateBackupLocation(provider, backupLocation, credName, bucketName, orgID)
+				CreateProviderClusterObject(provider, kubeconfigList, credName, orgID)
+			}
+		})
+
+		// Moment in time when tests should finish
+		end := time.Now().Add(time.Duration(Inst().MinRunTimeMins) * time.Minute)
+
+		for time.Now().Before(end) {
+			Step("Deploy applications", func() {
+				for _, provider := range providers {
+					providerClusterConfigPath, err := getProviderClusterConfigPath(provider, kubeconfigList)
+					Expect(err).NotTo(HaveOccurred(),
+						fmt.Sprintf("Failed to get kubeconfig path for provider %s cluster. Error: [%v]", provider, err))
+					logrus.Infof("Set context to %s", providerClusterConfigPath)
+					SetClusterContext(providerClusterConfigPath)
+
+					providerContexts := make([]*scheduler.Context, 0)
+					providerNamespaces := make([]string, 0)
+
+					// Rescan specs for each provider to reload provider specific specs
+					logrus.Infof("Rescan specs for provider %s", provider)
+					err = Inst().S.RescanSpecs(Inst().SpecDir, provider)
+					Expect(err).NotTo(HaveOccurred(),
+						fmt.Sprintf("Failed to rescan specs from %s for storage provider %s. Error: [%v]",
+							Inst().SpecDir, provider, err))
+
+					logrus.Infof("Start deploy applications for provider %s", provider)
+					for i := 0; i < Inst().ScaleFactor; i++ {
+						taskName := fmt.Sprintf("%s-%s-%d", taskNamePrefix, provider, i)
+						logrus.Infof("Task name %s\n", taskName)
+						appContexts := ScheduleApplications(taskName)
+						providerContexts = append(providerContexts, appContexts...)
+
+						for _, ctx := range appContexts {
+							namespace := GetAppNamespace(ctx, taskName)
+							providerNamespaces = append(providerNamespaces, namespace)
+						}
+					}
+
+					contexts[provider] = providerContexts
+					bkpNamespaces[provider] = providerNamespaces
+				}
+			})
+
+			Step("Validate applications", func() {
+				for _, provider := range providers {
+					providerClusterConfigPath, err := getProviderClusterConfigPath(provider, kubeconfigList)
+					Expect(err).NotTo(HaveOccurred(),
+						fmt.Sprintf("Failed to get kubeconfig path for provider %s cluster. Error: [%v]", provider, err))
+					SetClusterContext(providerClusterConfigPath)
+
+					// In case of non-portworx volume provider skip volume validation until
+					// other volume providers are implemented.
+					for _, ctx := range contexts[provider] {
+						ctx.SkipVolumeValidation = true
+						ctx.ReadinessTimeout = BackupRestoreCompletionTimeoutMin * time.Minute
+					}
+
+					logrus.Infof("validate applications for provider %s", provider)
+					ValidateApplications(contexts[provider])
+				}
+			})
+
+			logrus.Info("Wait for IO to proceed\n")
+			//time.Sleep(time.Minute * 5)
+
+			// Perform all backup operations concurrently
+			// TODO(stgleb): Add multi-namespace backup when ready in px-backup
+			for _, provider := range providers {
```

Comment:
Can we parallelize backups accross providers? meaning in multiple goroutines we start backup for AWS, GKE and Azure clusters at same time.

Reply:
Well, the problem of concurrent execution on provider level is backupDriver it self. This is a singleton and SetClusterContext changes it state. Since we have cluster per each cloud provider I can't switch to another cluster before complete spawning backups for another one.
I don't think this is a real problem since backupussually takes minutes, so all backup request will be fire before first one completes which is actual purpose of scale tests.

Reply:
Got it